File: clients/go/model/order.go
package model

import "io"
import . "komus24/stream"

// Player's orders
type Order struct {
    // TODO - Document
    Vehicles []VehicleOrder
}

func NewOrder(vehicles []VehicleOrder) Order {
    return Order {
        Vehicles: vehicles,
    }
}

// Read Order from reader
func ReadOrder(reader io.Reader) Order {
    var vehicles []VehicleOrder
    vehicles = make([]VehicleOrder, ReadInt32(reader))
    for vehiclesIndex := range vehicles {
        var vehiclesElement VehicleOrder
        vehiclesElement = ReadVehicleOrder(reader)
        vehicles[vehiclesIndex] = vehiclesElement
    }
    return Order {
        Vehicles: vehicles,
    }
}

// Write Order to writer
func (order Order) Write(writer io.Writer) {
    vehicles := order.Vehicles
    WriteInt32(writer, int32(len(vehicles)))
    for _, vehiclesElement := range vehicles {
        vehiclesElement.Write(writer)
    }
}

// Get string representation of Order
func (order Order) String() string {
    stringResult := "{ "
    stringResult += "Vehicles: "
    vehicles := order.Vehicles
    stringResult += "[ "
    for vehiclesIndex, vehiclesElement := range vehicles {
        if vehiclesIndex != 0 {
            stringResult += ", "
        }
        stringResult += vehiclesElement.String()
    }
    stringResult += " ]"
    stringResult += " }"
    return stringResult
}

File: clients/python/model/order.py
from model.vehicle_order import VehicleOrder
from stream_wrapper import StreamWrapper
from typing import List

class Order:
    """Player's orders"""

    __slots__ = ("vehicles",)

    vehicles: List[VehicleOrder]

    def __init__(self, vehicles: List[VehicleOrder]):
        self.vehicles = vehicles
        """Orders for each of the vehicles"""

    @staticmethod
    def read_from(stream: StreamWrapper) -> "Order":
        """Read Order from input stream
        """
        vehicles = []
        for _ in range(stream.read_int()):
            vehicles_element = VehicleOrder.read_from(stream)
            vehicles.append(vehicles_element)
        return Order(vehicles)
    
    def write_to(self, stream: StreamWrapper):
        """Write Order to output stream
        """
        stream.write_int(len(self.vehicles))
        for element in self.vehicles:
            element.write_to(stream)
    
    def __repr__(self):
        return "Order(" + \
            repr(self.vehicles) + \
            ")"